languages/python/concepts/strings/about.md

```diff
@@ -1 +1,13 @@
-TODO: add information on strings concept
+Python strings can be powerful. At their core, a string is an iterable of Unicode code points. There is no separate "character" type in Python. For a deep dive on what information a string is encoding (or, "how does the computer know how to translate zeroes and ones into letters?"), [this blog post is enduringly helpful][joel-on-text].
+
+As a Python data type, strings can be manipulated by various methods, split into letters, used to join a list of strings into one string, amongst other things. The simplest way to create a string is by literally delimiting with `"` or `'` (e.g. `mystring = "astring"`); however, strings can also be written across multiple lines by using triple quotes (`"""`), or written to be used as a format string, like this: `f"{variable_will_be_interpolated_here}"`.
+
```

Comment:
```suggestion

### Construction  

The simplest way to create a `str` literal is by delimiting it with `"` or `'`.  Strings can also be written across multiple lines by using triple quotes (`"""` or `'''`) .   

```python
single_quoted = 'Single quotes allow "double quoting" without "escape" characters.'

double_quoted = "Double quotes allow embedded 'single quoting' without 'escape' characters".

triple_quoted =  '''Three single quotes or double quotes in a row allow for multi-line string literals.  You will most often encounter these as "doc strings" or "doc tests" written just below the first line of a function or class definition.  They are often used with auto documentation tools.''' 
```

String literals that are part of a single expression and are separated only by white space are _implicitly concatenated_ into a single string literal:

```python
("I do not "
"like "
"green eggs and ham.") == "I do not like green eggs and ham."
```  

Additionally,  [interpolated](https://en.wikipedia.org/wiki/String_interpolation) strings (`f-strings`) can be formed:

```python
my_name = "Praveen"

intro_string = f"Hi! My name is {my_name}, and I'm happy to be here!"

>>>print(intro_string)
Hi! My name is Praveen, and I'm happy to be here!
```

Finally, the  [`str()` constructor](https://docs.python.org/3/library/stdtypes.html#str) can be used to create/coerce strings from other objects/types.  However, the `str` constructor _**will not iterate**_ through an object , so if something like a `list` of elements needs to be connected, `.join()` is a better option: 

```python
>>> my_number = 675
>>> str(my_number)
'675'

>>> my_list = ["hen", "egg", "rooster"]
>>> str(my_list)
"['hen', 'egg', 'rooster']"

>>> ' '.join(my_list)
'hen egg rooster'
```

### Formatting

Python provides a rich set of tools for [formatting](https://docs.python.org/3/library/string.html#custom-string-formatting) and [templating](https://docs.python.org/3/library/string.html#template-strings) strings, as well as more sophisticated text processing through the [re (_regular expressions_)](https://docs.python.org/3/library/re.html), [difflib (_sequence comparison_)](https://docs.python.org/3/library/difflib.html), and [textwrap](https://docs.python.org/3/library/textwrap.html) modules.   For a great introduction to string formatting in Python, see [this post at Real Python](https://realpython.com/python-string-formatting/).  

For more details on string methods, see [Strings and Character Data in Python](https://realpython.com/python-strings/) at the same site.

###  Related types and encodings

In addition to `str` (a *text* sequence), Python has corresponding  [binary sequence types](https://docs.python.org/3/library/stdtypes.html#binaryseq) `bytes` (a *binary* sequence),  `bytearray` and `memoryview` for the efficient storage and handling of binary data.  Additionally, [Streams](https://docs.python.org/3/library/asyncio-stream.html#streams) allow sending and receiving binary data over a network connection without using callbacks.
```